### PR TITLE
fix an override logic in Inherit function

### PR DESCRIPTION
**Pull request by karta0807913**
Signed-off-by: karta0807913 <karta0807913@gmail.com>

Refer to issue #16632

#### Does this PR introduce a user-facing change?

No

```release-note
The variable `s` was unmarshaled from `compatibleOptions`, but `libpod.InfraInherit` doesn't have the `ipcns` field.
Therefore, s.IpcNS.NSMode is always empty.
```

### CODE REVIEW

**Comment by giuseppe** [comment]
LGTM, could you add a test?

**Comment by rhatdan** [comment]
@karta0807913 Either add a test or [NO NEW TESTS NEEDED] to the message above and then force push so we can merge.
LGTM
/approve

**Comment by karta0807913** [comment]
Sure, give me some time to figure out how to add a test :D

File: pkg/specgen/generate/kube/kube.go [hunk 1]
```diff
@@ -145,6 +146,21 @@ type CtrSpecGenOptions struct {
...
+
+	if s.CgroupsMode == "" {
+		s.CgroupsMode = rtc.Cgroups()
+	}
+	if len(s.ImageVolumeMode) == 0 {
+		s.ImageVolumeMode = rtc.Engine.ImageVolumeMode
+	}
+	if s.ImageVolumeMode == "bind" {
+		s.ImageVolumeMode = "anonymous"
+	}
```
Comment (karta0807913) [inline pkg/specgen/generate/kube/kube.go:162]: @giuseppe could you check the logic here?
I copy them from the fill spec function.

**Comment by karta0807913** [comment]
HI @rhatdan, I fix all tests.
Could you check my code again?
Thank you!

**Pull request merged**